### ISSUE TITLE
Spies are auto-deploying

### ISSUE THREAD

**Issue by gdwitt** [LordGolias/Brytenwalda#16]
I wish I could put the pic here. After 1 week into game, most players are reporting that they are getting spy reports from their "spies" that were never deployed or hired. There are several screenshots on the forums. 
Last night, I pulled up the original Brytenwalda source files (1.40) where this was not a problem and tried to compare the spy scripts line by line with the Bryt Reworked v23 source.
I could not find any differences other than the wages were increased for being on mission.
I used search terms like "spies" "spy" spies_deployed and slot_spy_target_town to identify anything new. I can't find anything. Perhaps there's is something sick with the slots?

Anyways, to make it less of a problem, I decreased the relationship drop with the Lords to 0 (since that's where it really hurts) and the relationship drop with towns to -1. If we removed it completely, I think it would be hard to see when this occurs in the rgl_log unless I add debug statements.

> (store_add,":kill_modified",":kill_chance",":total_relation"),
> 							(try_begin),
> 								(le,":kill_modified",15),
> 								(party_set_slot,":cur_center",slot_spy_in_town,0),
> 								(party_set_slot,":cur_center",slot_spy_sabotage,0),
> 								(val_sub,":spies_deployed",1),
> 								(party_set_slot,"p_main_party",slot_spies_deployed,":spies_deployed"),
> 								(party_get_slot, ":lord_troop_id", ":cur_center", slot_town_lord),
> 								(call_script, "script_change_player_relation_with_troop", ":lord_troop_id", -3),
> 								(call_script, "script_change_player_relation_with_center", ":cur_center", -5),
> 								(dialog_box,"str_spy_captured"),

**Comment by LordGolias** [comment]
Related to #6.

**Comment by LordGolias** [comment]
I've found the issue.

It is a slot issue, as I suspected, but it is not issue #6 . 

The issue is that the slot `slot_town_trade_good_prices_begin` requires allocating extra slots after it correspondent to the number of different goods defined by a constant. However, that number (currently ~50) is more than the difference between `slot_town_trade_good_prices_begin` and `slot_spy_in_town`.

Therefore, every time there is an update to the prices (once a week), the slot `slot_spy_in_town` gets populated with the price of a good on each center, which triggers a huge amount of events spy events.

Slot-slot collisions is a problem that I'm solving in this version of the code in a more general way, but in case anyone wants to apply the fix to their own code, the solution is to set `slot_town_trade_good_prices_begin` to be a number whose it + 100 does not collide with any other party slot.
